core/ui.js

```diff
@@ -383,6 +437,13 @@ document.addEventListener('isInterceptionWorking', function (e)
 {
     isInterceptionWorking = e.detail;
     deletedDB = indexedDB.open("deletedMsgs", 1)
+
```

Comment:
But if you use a global variable `theme` that's getting updated inside `isInterceptionWorking`, changing theme in the settings won't catch it. Maybe just create a function `getTheme` that queries the theme instead

Reply:
Ah I see, I have committed a new change. Everytime a new mutation (list of mutations) is observed, it will call `getTheme()` to check the theme

Reply:
Hmm, why not just call it inside `restoreDeletedMessage` instead of in every mutation?

Reply:
I was worried about performance issues in the case of the theme being `null` since it will have to check the CSS whether it is dark/light mode for every messageNode in the array.

But come to think of it, calling it for every array of mutation is not exactly a lot better either.

Reply:
Oh OK